src/components/TextReuseClustersPage.vue

```diff
@@ -134,25 +132,41 @@ export default {
     handleSearchInputSubmitted(searchText) {
       if (searchText === '') return
       this.$navigation.updateQueryParametersWithHistory({
-        [QueryParameters.SearchText]: searchText
+        [QueryParameters.SearchText]: searchText,
+        [QueryParameters.PageNumber]: 1,
       })
     },
     handlePaginationPageChanged(page) {
       this.$navigation.updateQueryParameters({
-        [QueryParameters.PageNumber]: page - 1
+        [QueryParameters.PageNumber]: page,
       })
     },
     handleOrderByChanged(orderByValue) {
       this.$navigation.updateQueryParameters({
-        [QueryParameters.OrderBy]: orderByValue
+        [QueryParameters.OrderBy]: orderByValue,
+        [QueryParameters.PageNumber]: 1,
       })
     },
     handleFiltersEnabledChanged(filtersAreEnabled) {
       this.$navigation.updateQueryParameters({
         [QueryParameters.SearchFiltersEnabled]: filtersAreEnabled ? 1 : undefined
       })
     },
-    isLastItem
+    isLastItem,
+    async loadClusters({ query }) {
+      this.isLoading = true;
+      [this.clusterItems, this.searchInfo] = await textReuseClusters
+        .find({ query })
+        .then(result => [result.clusters, result.info]);
+      console.info('loadClusters() success: ', this.searchInfo);
+      // updte pagination list.
+      this.paginationList = {
```

Comment:
`paginationList` is totally asking to be made a computed property. When we set `currentPage` here it becomes potentially out of sync with query parameters and then will not be used when the query is constructed in `searchApiQueryParameters` next time.